### DISCUSSION TITLE
Problem when moving _targets  from Unix to Windows

### DISCUSSION THREAD

**Discussion by shambhu112** [ropensci/targets#459] [Help]
## Issue
I have a targets repository folder I have created on ubuntu . When I export the rstudio folder project as a zip and run this folder in windows I get the following error : Error in gzfile(file, "rb") : invalid 'description' argument    This error happens at the time of 

`targets::tar_read_raw("bank_2020_kpis")`  I have tried `targets::tar_load("bank_2020_kpis")`  also.

Is this is a known issue between unix and windows ?  

> sessionInfo:= on ubuntu
` R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.2 LTS
> Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0`

I can update this issue with a recreatable example. It will take me a few days

**Comment by wlandau** [comment]
It's hard to say at the moment, but here are some possible explanations:

* The files `_targets/objects/bank_2020_kpis` and/or `_targets/meta/meta` may be missing.
* You might have different R versions or package versions on Ubuntu vs Windows.
* You may be running R in a working directory other than the project root. In order to read the target, `list.files(getwd())` must include `"_targets"`.

The error "Error in gzfile(file, "rb") : invalid 'description' argument" usually comes up during `readRDS()` when some kind of output is missing or invalid. So one thing you could try is `readRDS("_targets/objects/bank_2020_kpis")` on both machines.

**Reply by shambhu112** [reply]
Thanks for your quick reply.  
a few updates
1.  both '_targets/objects/bank_2020_kpi' and '_targets/meta/meta'  exists
2.  Running R in the right Working Directory
3. readRDS("_targets/objects/bank_2020_kpis")  works well.  

Let me dig into this more and get back.  At this time, I will probably do the workaround of 'readRDS("_targets/objects/bank_2020_kpis")' and move forward

**Comment by shirdekel** [comment]
I'm also having this issue between two Macs (both 10.14.6). The files are shared between them using Dropbox.

The call to `targets::tar_read(object_name)` works for one computer, but for the other I get `Error in gzfile(file, "rb") : invalid 'description' argument`.

For both computers:
- `readRDS("_targets/objects/object_name")` works
- `_targets/meta/meta` exists
- `list.files(getwd())` includes "_targets"
- `targets::tar_make()` runs fine

**Reply by wlandau** [reply]
Any chance you could post an example so I can try it out and debug?

**Reply by shirdekel** [reply]
I'll have a go at a minimal reprex when I'm at both computers again.

**Reply by shirdekel** [reply]
I think I got it.

The issue was caused by different `{targets}` versions. Computer 1 had 0.4.2.9000 and Computer 2 (the one with the error) had 0.2.0. Updating `{targets}` to the most recent development version in Computer 2 fixed the issue. Looking at `_targets/meta/process` helped me diagnose this.

I'm not sure what the actual core issue was, but it was likely something to do with the way that the object was being coded in the old version within `_targets/meta/meta`. Specifically:

`test|stem|f9fd3d0282f65322|a6f445873b6f674c|ef46db3751d8e999|1064684737|_targets/objects/test|c5f705a1ed080d4f|9064907978062c20|56|rds|vector|||0.003||`

instead of 

`test|stem|f9fd3d0282f65322|a6f445873b6f674c|ef46db3751d8e999|1064684737||t18806.9611826675s|9064907978062c20|56|rds|vector|||0.003||`

See this repo and its git history for an example: https://github.com/shirdekel/targets-test

**Reply by wlandau** [reply]
Thanks for the example. I ran `tar_make()` on https://github.com/shirdekel/targets-test/blob/master/_targets.R using 0.2.0, then successively again on 0.2.0, 0.3.0, 0.3.1, 0.4.1, 0.4.2, and 0.5.0.9002. After the first `tar_make()`, each successive `tar_make()` correctly skipped the target. So maybe it has something to do with other parts of your package environment? What if you used an `renv` library to hold all your packages constant across computers? And what about R versions?

The metadata you mention changed because of https://github.com/ropensci/targets/pull/407, but I thought I did that compatibly. (Looks that way locally on my machine at any rate.) 

What does `tar_sitrep()` say when the target is incorrectly invalidated? What cues are activated?

**Reply by shirdekel** [reply]
The way I reproduce it is by installing version 0.2.0 on Computer 2, running `tar_make()` on Computer 1 (which has version, 0.4.2.9000) and then calling `tar_read(test)` on Computer 2.

I assumed it was a `targets` issue because I didn't use `renv` to reproduce this and R versions stayed constant.

`tar_sitrep()` gives the following in Computer 2 after I see the error:

```r
# A tibble: 1 x 9
  name  record always never command depend format iteration file 
  <chr> <lgl>  <lgl>  <lgl> <lgl>   <lgl>  <lgl>  <lgl>     <lgl>
1 test  FALSE  FALSE  FALSE FALSE   FALSE  FALSE  FALSE     TRUE 
```

**Reply by shirdekel** [reply]
But actually, I'm struggling to reproduce it if I do the same procedure within one computer. So you might be right about it being something else.

**Reply by wlandau** [reply]
> The way I reproduce it is by installing version 0.2.0 on Computer 2, running tar_make() on Computer 1 (which has version, 0.4.2.9000) and then calling tar_read(test) on Computer 2.

That actually makes sense. If you run the pipeline with `targets` >= 0.4.2, then `targets` <= 0.3.1 will not be able to read the data because of #407. Upgrading `targets` on Computer 2 should solve that the issue.

As of #407, the metadata no longer records file paths unless you use `format = "file"` or `format = "aws_file"` in `tar_target()` or `tar_option_set()`. That's why 0.2.0 cannot read from newer data stores.

**Reply by shirdekel** [reply]
Ah ok. Thanks for looking into it!
